app/views/favours/show.html.erb

```diff
@@ -15,9 +15,19 @@
 <p>Description: <%= @favour.description %></p>
 
 <% if user_signed_in? %>
-	<% unless current_user.applied?(@favour) %>
-	  <a href="#" class="btn btn-primary">Apply [Pending - Favour Application]</a>
-	<% end %>
+	<!-- Apply for favour -->
+	<% unless current_user.applied?(@favour) || @favour.recipient == current_user %>
+    <a href="#" class="btn btn-primary">Apply [Pending - Favour Application]</a>
+  <% end %>
+  <!-- Already applied for favour -->
+  <% if current_user.applied?(@favour) %>
+    <p class="badge badge-success">You've already applied!</p>
+  <% end %>
+  <!-- Edit favour -->
+  <% if @favour.recipient == current_user %>
```

Comment:
We'll need to add the same check as the favours index page to make sure no one has applied.

Reply:
Added it into the show page as well now